src/js/tabs/setup.js

```diff
@@ -595,6 +597,36 @@ setup.cleanup = function (callback) {
     if (callback) callback();
 };
 
+setup.updateGyroStatus = function (armingDisableFlags) {
+    const NO_GYRO_BIT = 0; // Position of NO_GYRO flag
+    const CALIBRATING_BIT = 12; // Position of CALIBRATING flag
+    const ACC_CALIBRATION_BIT = 23; // Position of ACC_CALIBRATION flag
+
+    const hasNoGyro = armingDisableFlags & (1 << NO_GYRO_BIT);
+    const isCalibrating =
+        armingDisableFlags & (1 << CALIBRATING_BIT) || armingDisableFlags & (1 << ACC_CALIBRATION_BIT);
+
+    const hasGyroIssue = hasNoGyro || isCalibrating;
+
+    if (hasGyroIssue) {
+        $(".overlay-status").show();
+        $("#canvas_wrapper").addClass("blur");
+
+        // Show appropriate text based on the issue
+        if (hasNoGyro) {
+            $(".overlay-status .status-message").text(i18n.getMessage("gyroErroredText"));
+            $(".overlay-status .status-description").text(i18n.getMessage("gyroErroredDescriptionText"));
+        } else {
+            $(".overlay-status .status-message").text(i18n.getMessage("accCalibratingMessageText"));
+            $(".overlay-status .status-description").text(i18n.getMessage("accCalibratingDescriptionText"));
+        }
+        $(".overlay-status").show();
+    } else {
+        $(".overlay-status").hide();
+        $("#canvas_wrapper").removeClass("blur");
+    }
```

Comment:
```suggestion
    if (hasGyroIssue) {
        // Show appropriate text based on the issue
        $(".overlay-status .status-message").text(i18n.getMessage(hasNoGyro ? "gyroErroredText" : "accCalibratingMessageText"));
        $(".overlay-status .status-description").text(i18n.getMessage(hasNoGyro ? "gyroErroredDescriptionText" : "accCalibratingDescriptionText"));
    }
    $("#canvas_wrapper").toggleClass("blur", hasGyroIssue);
    $(".overlay-status").toggle(hasGyroIssue);
```

Reply:
I tried already the toggle approach but is not working as expected because this is being called all the time.

https://github.com/user-attachments/assets/5bb841ce-85b2-473e-9b31-847ebe4c2108